app/src/lib/git/branch.ts

```diff
@@ -65,54 +64,50 @@ export async function deleteLocalBranch(
 }
 
 /**
- * Delete the branch. If the branch has a remote branch and `includeRemote` is true, it too will be
- * deleted. Silently deletes local branch if remote one is already deleted.
+ * Deletes a remote branch
+ *
+ * @param remoteName - the name of the remote to delete the branch from
+ * @param remoteBranchName - the name of the branch on the remote
+ * @param branchTipSha - the branch's sha is logged for reference, more useful
+ * for a remote only branch if the user wants to undo this delete.
  */
-export async function deleteBranch(
+export async function deleteRemoteBranch(
   repository: Repository,
-  branch: Branch,
   account: IGitAccount | null,
-  includeRemote: boolean
+  remoteName: string,
+  remoteBranchName: string,
+  branchTipSha: string
 ): Promise<true> {
-  if (branch.type === BranchType.Local) {
-    await deleteLocalBranch(repository, branch.name)
+  const networkArguments = await gitNetworkArguments(repository, account)
+  const remoteUrl =
+    (await getRemoteURL(repository, remoteName).catch(err => {
+      // If we can't get the URL then it's very unlikely Git will be able to
+      // either and the push will fail. The URL is only used to resolve the
+      // proxy though so it's not critical.
+      log.error(`Could not resolve remote url for remote ${remoteName}`, err)
+      return null
+    })) || getFallbackUrlForProxyResolve(account, repository)
+
+  const args = [...networkArguments, 'push', remoteName, `:${remoteBranchName}`]
+
+  // If the user is not authenticated, the push is going to fail
+  // Let this propagate and leave it to the caller to handle
+  const result = await git(args, repository.path, 'deleteRemoteBranch', {
+    env: await envForRemoteOperation(account, remoteUrl),
+    expectedErrors: new Set<DugiteError>([DugiteError.BranchDeletionFailed]),
+  })
+
+  if (branchTipSha !== undefined) {
+    log.info(`Deleted branch ${remoteBranchName} (was ${branchTipSha})`)
```

Comment:
Looking at the rest of the codebase, it doesn't look like we're leaving many traces for debugging, so I think I would remove this since:
1. It's logging a successful case for a very probable scenario (I see more useful logging errors in this case, which we are probably already doing in the git invocation).
2. It forces us to add `branchTipSha` as a parameter, just for logging purposes 😕 

Is there any good reason to keep this log line and "pollute" the function signature with `branchTipSha`? 😳

Reply:
I agree it does pollute the function signature but it does have valid purpose (maybe there is a better approach). 

If a user deletes a remote branch on github.com, they can undo it. @niik informed me during pairing session, that a user can also restore a remote branch if they know the branches sha. By providing the sha in the log, we provide a way that the user can restore a mistakenly deleted remote branch, which is also an effort to provide similar functionality to github.com.

I believe it is important to keep it. It also seems logical to place in the `deleteRemoteBranch` to ensure it is captured with that action, but another option for us is to back it out to the `app-store.ts` `deleteBranch` method. What do you think about that?

Reply:
> but another option for us is to back it out to the `app-store.ts` `deleteBranch` method. What do you think about that? 

I think that would bother me less, yeah 😳

Reply:
Done :)